juniorguru_chick/lib/threads.py

```diff
@@ -23,20 +24,41 @@ async def fetch_starting_message(thread: discord.Thread) -> discord.Message | No
     except discord.errors.NotFound:
         return None
 
+def name_thread(message: discord.Message, name_template) -> str | None:
+
+    brackets_regex = re.compile(r"""
+        ^\[     # starts with [
+        .*      # any number of any characters
+        [^\s]   # not a whitespace
+        [^\]]   # not a closing bracket
+        \]      # ends with ]
+        """, re.VERBOSE)
+
+    """If the message includes text in square brackets, use that as name for the thread. Otherwise, use the name template."""
+    if (match:= re.match(brackets_regex, message.content)) is not None:
+        content = match.group()[1:-1]
```

Comment:
If you add parentheses to the regex, you'll be able to get it like this:

```python
r'bla bla bla (foo bar) bla'

.group(0) # bla bla bla foo bar bla
.group(1) # foo bar
```

You can even name them:


```python
r'bla bla bla (?P<iloveregexs>foo bar) bla'

.group(0) # bla bla bla foo bar bla
.group(1) # foo bar
.group('iloveregexs') # foo bar
```

Reply:
Ah, thank you for explaining this! I looked at the example which you provided previously and went "how do I even read this". My bad, I should have spent more time with it.